include/aspect/introspection.h

```diff
@@ -188,6 +188,17 @@ namespace aspect
        */
       const BaseElements base_elements;
 
+      struct Degrees
+      {
+        unsigned int       velocities;
+        unsigned int       temperature;
+        unsigned int       compositional_fields;
+      };
+      /**
+       * A variable that enumerates the base elements of the finite element
+       * that correspond to each of the variables in this problem.
```

Comment:
Yes, it was copied from other place. The new commit 9a1c710 addressed all  you early comments.